mirgecom/multiphysics/thermally_coupled_fluid_wall.py

```diff
@@ -256,20 +256,38 @@ def __init__(
         self._t_plus = t_plus
         self._grad_t_plus = grad_t_plus
 
-    def kappa_plus(self, dcoll, dd_bdry):
+    def proj_kappa_plus(self, dcoll, dd_bdry):
+        # project kappa plus in case of overintegration
+        if isinstance(self._kappa_plus, np.ndarray):
+            # orthotropic materials
+            actx = self._t_plus.array_context
+            normal = actx.thaw(dcoll.normal(dd_bdry))
+            kappa_plus = project_from_base(dcoll, dd_bdry, self._kappa_plus)
+            return np.dot(normal, kappa_plus*normal)
         return project_from_base(dcoll, dd_bdry, self._kappa_plus)
 
+    def proj_kappa_minus(self, dcoll, dd_bdry, kappa):
+        # state minus is already in the quadrature domain
+        if isinstance(kappa, np.ndarray):
+            # orthotropic materials
+            actx = self._t_plus.array_context
+            normal = actx.thaw(dcoll.normal(dd_bdry))
+            return np.dot(normal, kappa*normal)
+        return kappa
+
```

Comment:
The biggest issue was that `temperature_bc` was returning with `shape=2` because of kappa's `shape=2`.

Lemme think on your comment, thanks.

Reply:
`temperature_bc` is still normal-dotted, similarly to diffusion.

The issue I was facing came exactly from the penalization, as you said above. Now I am normal-dotting only for the penalization (https://github.com/illinois-ceesd/mirgecom/pull/1009/commits/b08199880eefb7b11d3df0e32c596bddefeaeb43)

Reply:
https://github.com/illinois-ceesd/mirgecom/pull/1009/commits/b48506ec80c6f01559ab1932181606b22ebbf958 fixes the kappa_plus in `kappa_bc`. It only projects the individual components now to the quadrature domain, instead of projecting the normal kappa.